### PR TITLE
Budget creation page

### PR DESCRIPTION
**Pull request by Skczu**

### CODE REVIEW

File: components/generic/Auth/Auth.tsx [hunk 1]
```diff
@@ -17,9 +17,9 @@ export const Auth = ({ children, forceRedirect = true }: AuthProps) => {
     }
   }, [forceRedirect, isLoading, logged, router]);
 
-  if (!logged) {
-    return null;
-  }
+  // if (!logged) {
+  //   return null;
+  // }
```
Comment (wisnie) [inline components/generic/Auth/Auth.tsx]: Why do you change generic component? Just don't wrap everything in `Auth`.

**Comment by wisnie** [comment]
We have wrong paths here, it should be `pages/communities/[communitySlug]` and `pages/communities/[communitySlug]/budgets/[budgetSlug]`.

File: components/composited/AccountPage/AccountPage.tsx [hunk 2]
```diff
@@ -13,6 +13,9 @@ export const AccountPage = () => {
 
   return (
     <Auth>
```
Comment (wisnie) [inline components/composited/AccountPage/AccountPage.tsx:15]: We might have created an `Not Authorized` page, but for mvp it's okay to redirect.

File: components/composited/BudgetForm/BudgetForm.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    e.preventDefault();
+
+    const formData = new FormData(e.currentTarget);
+
+    const result = BudgetSch.safeParse({
+      name: formData.get('name'),
+      description: formData.get('description'),
+      budgetType: formData.get('budgetType'),
+      estimatedCost: Number(formData.get('estimatedCost')),
+      estimatedRealizationDate: formData.get('estimatedRealizationDate'),
```
Comment (wisnie) [inline components/composited/BudgetForm/BudgetForm.tsx]: We will probably create simple utility function that will handle convertion to currency format.

File: components/composited/BudgetForm/BudgetForm.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      coordinator: user?.id,
+    });
+
+    if (result.success) {
+      onSubmit(result.data);
+      setInputErrorMessage(false);
+      return;
+    }
+
+    console.log(result.error);
```
Comment (wisnie) [inline components/composited/BudgetForm/BudgetForm.tsx]:
```suggestion
```

File: components/composited/CreateBudgetPage/CreateBudgetPage.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        as="h1"
+        content="Nowy budżet"
+        variant="base"
+        displayDecorationBorder={true}
+      />
+      {createNewBudget.isError && (
+        <ErrorMessage
+          className="mb-6 sm:col-end-2"
+          title="Nie udało się zalogować!"
+          description="Wystąpił błąd wprowadzonych danych, sprawdź ich poprawność. Jeśli nie możesz rozwiązać problemu, skontakuj się z administracją serwisu."
```
Comment (wisnie) [inline components/composited/CreateBudgetPage/CreateBudgetPage.tsx]: If we reuse the same error name, let's move it to constant as a `const GENERIC_ERROR_NAME`.

File: components/generic/Auth/Auth.tsx (hunk 1 above)
Comment (wisnie) [inline components/generic/Auth/Auth.tsx]: Please enable those lines.

File: components/generic/Select/Select.tsx [hunk 6]
```diff
@@ -9,25 +8,23 @@ type Option = {
 type SelectProps = {
   name: string;
   label: string;
-  value: string;
   options: Option[];
-  onChange: ChangeEventHandler<HTMLSelectElement>;
 };
 
-export const Select = ({ name, label, value, options, onChange }: SelectProps) => {
+export const Select = ({ name, label, options }: SelectProps) => {
```
Comment (wisnie) [inline components/generic/Select/Select.tsx]: Those props will be used and they are necessary to make this component generic, add them.

File: validators/BudgetFields.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+export const BudgetSch = z.object({
+  name: z.string(),
+  description: z.string(),
+  budgetType: z.enum(['static', 'progressive']),
+  estimatedCost: z.number(),
+  // Zod docs recommended way of accepting dates as strings
+  estimatedRealizationDate: z.preprocess((arg) => {
+    if (typeof arg == 'string' || arg instanceof Date) return new Date(arg);
+  }, z.date()),
```
Comment (wisnie) [inline validators/BudgetFields.ts]: Is it in ISOString format?

**Review by wisnie — CHANGES_REQUESTED** [review]
Some things have to be changed.

File: validators/BudgetFields.ts (hunk 7 above)
Comment (Skczu) [reply validators/BudgetFields.ts]: Date string from input is passed into Date constructor and it works with the DB, tested.

**Comment by wisnie** [comment]
@wisnie I'll take it from tomorrow.

**Comment by wisnie** [comment]
As for currencies, we will use type string with regex that will match only integers.

**Pull request merged**